core/server/services/routing/bootstrap.js

```diff
@@ -38,7 +39,7 @@ module.exports.init = (options = {start: false}) => {
  *   - is the PreviewRouter an app?
  */
 module.exports.start = () => {
-    const apiVersion = themeService.getActive().engine('ghost-api');
+    const apiVersion = themeService.getActive() ? themeService.getActive().engine('ghost-api') : defaultThemeEngine['ghost-api'];
```

Comment:
In general: i think the url and routing service need to initialise if the active theme can't be found. The direction looks good 👍 

---

I think requiring a specific file from the theme service is quite "dirt"y, because we don't use the official theme interface? It's quite weird that "if there is no active theme, that we require a file from the theme service to get a default"?

Reply:
Hmm I agree its a bit dirty fetching the default config file, this is a bit of unique case where instead of not having particular value in theme interface, we don't have the theme interface at all, so there is nothing directly to fallback to. Since what we are trying to fetch is engine for a non-existent theme here, themeService seems to be the next best contender to answer it. 

So as fallback the only choices seemed to be between hard-coding a value and using defaults file, what went in favour of latter was -
- Fallback value stays consistent everywhere, picked from one place
- This is the default value we use for a theme if it doesn't specify `ghost-api`, so it seemed like a natural extension in case of no theme itself. 

The only other option I see for getting a fallback is if we tamper theme interface to load defaults instead of `null` in case of non-existent theme, but that seems stretching it a bit too much for the values, wdyt?

Reply:
In general: The underlying problemis that the API is currently tied with the url generation.
Content API returns urls. That's why we have to generate the urls even if there is no active theme.

Maybe we could just extend the theme service interface _for now_ like

```
if themeService.hasActiveTheme()
  apiVersion = getActive().engine()
else
  apiVersion = themeService.getDefaults().api_version
```

Reply:
I like this direction, extending themeService to have `getApiVersion` makes much more sense as it's used across a lot of files which will be much cleaner instead of all the `if/else` everywhere. 👍

Reply:
@kirrg001 Updated. Looks much cleaner now